google/resource_storage_bucket.go

```diff
@@ -378,3 +448,135 @@ func flattenCors(corsRules []*storage.BucketCors) []map[string]interface{} {
 	}
 	return corsRulesSchema
 }
+
+func resourceGCSBucketLifecycleCreateOrUpdate(d *schema.ResourceData, sb *storage.Bucket, update bool) error {
+	if v, ok := d.GetOk("lifecycle_rule"); ok {
+		lifecycle_rules := v.([]interface{})
+
+		if len(lifecycle_rules) > 100 {
+			return fmt.Errorf("At most 100 lifecycle_rule blocks are allowed")
```

Comment:
Would be nice to print how many `lifecycle_rules` were actually found in the error message